{
    "url": "https://api.github.com/repos/spack/spack/issues/9392",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/9392/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/9392/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/9392/events",
    "html_url": "https://github.com/spack/spack/issues/9392",
    "id": 365549799,
    "node_id": "MDU6SXNzdWUzNjU1NDk3OTk=",
    "number": 9392,
    "title": "Infinite recursion getattr",
    "user": {
        "login": "ibaned",
        "id": 971921,
        "node_id": "MDQ6VXNlcjk3MTkyMQ==",
        "avatar_url": "https://avatars.githubusercontent.com/u/971921?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/ibaned",
        "html_url": "https://github.com/ibaned",
        "followers_url": "https://api.github.com/users/ibaned/followers",
        "following_url": "https://api.github.com/users/ibaned/following{/other_user}",
        "gists_url": "https://api.github.com/users/ibaned/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/ibaned/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/ibaned/subscriptions",
        "organizations_url": "https://api.github.com/users/ibaned/orgs",
        "repos_url": "https://api.github.com/users/ibaned/repos",
        "events_url": "https://api.github.com/users/ibaned/events{/privacy}",
        "received_events_url": "https://api.github.com/users/ibaned/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 3,
    "created_at": "2018-10-01T16:51:10Z",
    "updated_at": "2018-10-01T18:13:41Z",
    "closed_at": "2018-10-01T18:13:41Z",
    "author_association": "CONTRIBUTOR",
    "active_lock_reason": null,
    "body": "While trying to run any `spack` command, I get an error about maximum recursion. It looks like `getattr` is calling itself in one of the Python classes.\r\n\r\n### Steps to reproduce the issue\r\n\r\n```console\r\n$ source $SPACK_ROOT/share/spack/setup-env.sh\r\n```\r\n\r\n### Error Message\r\n\r\n```console\r\n...\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 556, in __getattr__\r\n    return getattr(self.instance, name)\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 556, in __getattr__\r\n    return getattr(self.instance, name)\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 556, in __getattr__\r\n    return getattr(self.instance, name)\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 556, in __getattr__\r\n    return getattr(self.instance, name)\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 556, in __getattr__\r\n    return getattr(self.instance, name)\r\n  File \"/home/daibane/src/spack/lib/spack/llnl/util/lang.py\", line 552, in instance\r\n    self._instance = self.factory()\r\n  File \"/home/daibane/src/spack/lib/spack/spack/config.py\", line 547, in _config\r\n    defaults = InternalConfigScope('_builtin', config_defaults)\r\n  File \"/home/daibane/src/spack/lib/spack/spack/config.py\", line 238, in __init__\r\n    _validate_section({section: dsec}, section_schemas[section])\r\n  File \"/home/daibane/src/spack/lib/spack/spack/config.py\", line 611, in _validate_section\r\n    _validate_section.validator(schema).validate(data)\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/validators.py\", line 67, in __init__\r\n    resolver = RefResolver.from_schema(schema)\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/validators.py\", line 260, in from_schema\r\n    return cls(schema.get(u\"id\", u\"\"), schema, *args, **kwargs)\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/validators.py\", line 245, in __init__\r\n    for id, validator in iteritems(meta_schemas)\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/_utils.py\", line 20, in __init__\r\n    self.store.update(*args, **kwargs)\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/validators.py\", line 244, in <genexpr>\r\n    (id, validator.META_SCHEMA)\r\n  File \"/usr/lib/python2.7/_abcoll.py\", line 406, in iteritems\r\n    yield (key, self[key])\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/_utils.py\", line 23, in __getitem__\r\n    return self.store[self.normalize(uri)]\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/_utils.py\", line 16, in normalize\r\n    return urlsplit(uri).geturl()\r\n  File \"/home/daibane/src/spack/lib/spack/external/jsonschema/compat.py\", line 37, in urlsplit\r\n    scheme, netloc, path, query, fragment = _urlsplit(url)\r\n  File \"/usr/lib/python2.7/urlparse.py\", line 174, in urlsplit\r\n    allow_fragments = bool(allow_fragments)\r\nRuntimeError: maximum recursion depth exceeded while calling a Python object\r\n```\r\n\r\n### Information on your system\r\n\r\nThis is an Ubuntu 17.10 system",
    "performed_via_github_app": null
}